File: leetcode_1346/CheckIfNandItsDoubleExist.java
package leetcode_1346;

import java.util.HashSet;
import java.util.Set;

class CheckIfNandItsDoubleExist{
    public boolean checkIfExist(int[] arr) {
       Set<Integer> set = new HashSet<>();
       for(int num : arr)
       {
            if(set.contains(num * 2) || (num % 2 == 0 && set.contains(num / 2)))
                return true;
            set.add(num);
       }
       return false;
    }

    public static void main(String[] args) {
        CheckIfNandItsDoubleExist obj = new CheckIfNandItsDoubleExist();
        int[] arr = {10, 2, 5, 3};
        System.out.println(obj.checkIfExist(arr));
    }
}

File: leetcode_1346/CheckIfNandItsDoubleExist.py
class CheckIfNandItsDoubleExist:
    def checkIfExist(self, arr: list[int]) -> bool:
        seen = set()
        
        for num in arr:
            if (num * 2 in seen) or (num % 2 == 0 and num // 2 in seen):
                return True
            seen.add(num)
        return False
    
def main():
    arr = [10,2,5,3]
    cind = CheckIfNandItsDoubleExist()
    print(cind.checkIfExist(arr))

if __name__ == "__main__":
    main()